### ISSUE TITLE
Add an argument to choose the template (blocking, async...)

### ISSUE THREAD

**Issue by SergioGasquez** [esp-rs/esp-generate#27]

**Comment by bjoernQ** [comment]
I guess it's worth to discuss this: Is this really something we need and want?

It's taken over from the original cargo-generate template where we haven't had support for async but now we can support multiple `main` files with one template.

We considered this to be used to support additional frameworks like RTIC (or ArielOS or whatever) but I'm not too confident we should even do that - i.e. if we have an option for it in _our_ generator users will expect at least some level of support by us

Maybe there is another use-case I am not thinking of?

**Comment by SergioGasquez** [comment]
> I guess it's worth to discuss this: Is this really something we need and want?

Agree, I dont think we need this anymore, current state is good imho. If we change our minds we can always reopen the issue!
